data/download_datasets.py

```
@@ -0,0 +1,177 @@
+    if not output_file.exists():
+        download_url(url, output_file)
+        print(f"✓ Downloaded to {output_file}")
+    else:
+        print(f"✓ Already exists: {output_file}")
+    
+    # Extract
+    extract_dir = vctk_dir / "VCTK-Corpus-0.92"
+    if not extract_dir.exists():
+        print("Extracting archive... (this may take several minutes)")
+        with zipfile.ZipFile(output_file, 'r') as zip_ref:
+            zip_ref.extractall(vctk_dir)
```

Ruff (0.14.13)

Uses of `tarfile.extractall()`

(S202)